flocker/provision/_digitalocean.py

```diff
@@ -0,0 +1,242 @@
+    # * http://digitalocean.uservoice.com/forums/136585-digitalocean/suggestions/2814988-give-option-to-use-the-droplet-s-own-bootloader # noqa
+    # So rather than upgrade, we'll need to have new task to install the kernel
+    # package (and headers) for the DO supported kernel.
+    # The Fedora droplet default is to use a kernel that's too old for our
+    # purposes.
+    # Our documentation describes how to select a newer (DO supported) kernel
+    # for this droplet.
+    # Unfortunately this operation is only supported in the DO v2 API.
+    # * http://digitalocean.uservoice.com/forums/136585-digitalocean/suggestions/5618546-add-the-ability-to-change-kernel-via-api # noqa
+    # * https://developers.digitalocean.com/#change-the-kernel
+    # But libcloud only supports the DO v1 API
+    # * https://www.digitalocean.com/community/questions/does-libcloud-work-with-digitalocean-s-v2-api # noqa
+    # * https://issues.apache.org/jira/browse/JCLOUDS-613
+    v2client = pyocean.DigitalOcean(access_token=token)
+    v2droplet = v2client.droplet.get(node._node.id)
+
+    kernel = set_latest_droplet_kernel(v2droplet)
+    version, distribution, architecture = kernel.version.rsplit('.', 2)
+    version, release = version.split('-', 1)
+    run(
+        username='root',
+        address=node.address,
+        commands=task_install_kernel(version=version, release=release,
+                                     distribution='fc20',
+                                     architecture='x86_64')
+    )
+
+    # libcloud doesn't support shutting down DO vms.
+    # XXX Create an issue.
+    retry_if_pending(v2droplet.shutdown)
+
+    # Sorry about this, but shutdown returns the following, indicating that the
+    # droplet has halted, but it still seems to require some time before
+    # powering on.
+    # {u'status': u'completed', u'resource_id': 3797602, u'region': u'ams3', u'completed_at': u'2015-01-15T20:52:36Z', u'started_at': u'2015-01-15T20:52:31Z', u'type': u'shutdown', u'id': 41364967, u'resource_type': u'droplet'} # noqa
+    time.sleep(30)
+
+    # libcloud doesn't support powering up DO vms.
+    # XXX Create an issue.
+    retry_if_pending(v2droplet.power_on)
+
+    # Finally run all the standard Fedora20 installation steps.
+    run(
+        username='root',
+        address=node.address,
+        commands=provision(
+            package_source=package_source,
+            distribution=node.distribution,
+        )
+    )
+    return node.address
+
+
+IMAGE_NAMES = {
+    # It'd be better to use image ID here, but the following code is currently
+    # written to lookup image names...which would normally be good for
+    # readability but which in the case DigitalOcean are pretty meaningless.
+    'fedora-20': '20 x64',
+}
+
+
+def location_by_slug(driver, location_slug):
+    """
+    Look up a DigitalOcean location by its short human readable "slug" code.
+
+    # XXX: ``libcloud.DigitalOceanDriver.list_locations`` discards the slug
```

Comment:
The names are subject to change, I think. SLUG is a reasonable compromise between stability and human readability.